### ISSUE TITLE
[MacOS + Android] "Check required certificates are present" failed, desktop is not connecting to phone

### ISSUE THREAD

**Issue by vganin** [facebook/flipper#2789]
## 🐛 Bug Report
I added Android SDK 0.105.0 to my app, installed and ran Flipper 0.105.0 on MacOS but it doesn't connect to app. "Currently connecting..." is there forever. Chrome log says:
```
Timed out establishing connection with "Яндекс.Карты" on "Pixel 3a - 11 - API 30". Verify that your client is connected to Flipper and that there is no error related to idb. undefined
```
<img width="400" src="https://user-images.githubusercontent.com/11139794/132489116-e5024233-eb83-47a3-9f40-71d17b8ee0d5.png"><img width="405" alt="Screenshot at Sep 08 13-13-48" src="https://user-images.githubusercontent.com/11139794/132491485-2c032911-18cd-4fa6-baf7-3db02eed3658.png">
<img width="885" alt="Screenshot at Sep 08 13-23-56" src="https://user-images.githubusercontent.com/11139794/132492864-6fb9b9a0-09f6-4ecc-98d8-ffb55a0a4a27.png">
<img width="400" src="https://user-images.githubusercontent.com/11139794/132490245-3a1b210d-8a84-4277-aaeb-0a16b2c32fcf.jpg">

I did some research and found out that the message in diagnostic Activity is because of missing files in `files/sonar`. Turned out there was missing just one file - `connection_config.json`. All cert files were there. I added it manually with content `{"deviceId":"my_device_id_from_adb_devices"}` and it started to work. Not sure why it didn't work automatically though. If I reinstall app the bug reproduces again.
<img width="547" alt="Screenshot at Sep 08 13-20-36" src="https://user-images.githubusercontent.com/11139794/132492416-59025a25-1605-4099-803b-04310bc73dbb.png">

## To Reproduce
Just normally follow a guide to setup Android SDK and desktop Flipper 0.105.0

## Environment
Tested on Google Pixel Android 11 arm64 and AVD Android 10 x86
MacOS Big Sur 11.2.3

**Comment by mweststrate** [comment]
Thanks for the detailed report! We're investigating how it can get lost

**Comment by mweststrate** [comment]
Should be fixed in 108 which will release in a bit. Let us know if that doesn't solve the issue, I'm not 100% sure if it was the same issue as the one we fixed.

**Comment by jschear** [comment]
👋  I've been hitting what appears to be the exact same issue using desktop app 0.109.0 and Android client library 0.105.0. Manually generating a `connection_config.json` also resolved it for me.

For anyone also looking for a workaround, here's a way to push the file without hitting permissions issues:
```
adb push connection_config.json /data/local/tmp/connection_config.json
adb shell run-as <your app id> cp /data/local/tmp/connection_config.json /data/data/<your app id>/files/sonar/connection_config.json 
```

**Comment by mweststrate** [comment]
cc @lblasa

**Comment by lblasa** [comment]
@jschear, @vganin we believe to have fixed the issue. Please check the latest release [v0.111.0](https://github.com/facebook/flipper/releases/tag/v0.111.0)

**Comment by vganin** [comment]
@lblasa it worked for me, thanks!

**Comment by mweststrate** [comment]
Sorry I forgot to update the issue! The issue has been resolved in version 0.110 (and higher) and requires updating the native iOS / Android SDK per instructions here: https://fbflipper.com/docs/getting-started/react-native#using-the-latest-flipper-sdk
